cog/core/sql_abstract.py

```
@@ -0,0 +1,187 @@
+        if data is None:
+            return None
+
+        record = UserRecord(
+            _protected=True, **data  # pyright: ignore[reportArgumentType]
+        )
+
+        for field in fields(record):
+            if record.is_unset(field.name):
+                raise Exception(  # pylint: disable=broad-exception-raised
+                    f"SQL is not return all fields. (`{field.name}`=`{getattr(record, field.name)}`)"
+                )
```

Ruff (0.8.2)

Create your own exception

(TRY002)